module-system/sov-modules-api/src/state/prover_storage.rs

```diff
@@ -7,13 +7,13 @@ use jmt::storage::{NodeBatch, TreeWriter};
 use jmt::{JellyfishMerkleTree, KeyHash, RootHash, Version};
```

Comment:
Is it possible to leave `storage.rs, prover_storage.rs, zk_storage.rs, tree_db.rs` in `sov-state`?

Reply:
It's possible, yes, we just need to make a couple of private components public

Reply:
I think the current change is on the minimal. I moved only the components that are required to live with the `WorkingSet`. The root of the problem is with the `StateMap`, that consumes the `WorkingSet`:

https://github.com/Sovereign-Labs/sovereign-sdk/blob/a1d6e7dac747288d478cdb4c390a7711cdf7c9b5/module-system/sov-state/src/containers/map.rs#L120-L128

And it is required by the `Storage` trait:

https://github.com/Sovereign-Labs/sovereign-sdk/blob/a1d6e7dac747288d478cdb4c390a7711cdf7c9b5/module-system/sov-state/src/storage.rs#L225-L232

And `Storage` is required by everything else that was moved.

Reply:
We can open a tracking issue to revisit this in the future; wdyt? To be fair, it seems to me that `sov-state` is a module of `sov-modules-api`

Reply:
Thanks @vlopes11 yeah so we have circular dependency here.  Too me the sov-state is a bit more lower level than the `sov-modules-api`  The good news is that `verify_proof` has fairly simple implementation and it is used only in one place. One solution would be to bring only the `verify_proof` to the `sov-module-api` but I agree it is out of scope of this PR.